docs/action-release.md

```diff
@@ -0,0 +1,60 @@
+# Design: GitHub Action Release Process
+
+## Context
+
+We vend a variety of GitHub actions from the ros-tooling organization.
+Til now we have not defined a consolidated process for their management.
+This means that there is some confusion about release tags, indecision around the final structure of their repositories, and difficulties around the production of build artifacts.
+
+## Goal
+
+Lay out a cohesive plan for repository structure, reliable and automatic generation of build artifacts, and a release tagging plan.
+
+## Design
+
+### Repository structure
+
+Design constraint: Publish ROS CI Actions to the GitHub Marketplace for discoverability.
+
+There was a consideration of combining several of the ROS CI actions into a single repository, similar to https://github.com/octokit/graphql-action, as requested in https://github.com/ros-tooling/roadmap/issues/5
+
+However, in order to release our Actions into the GitHub Marketplace, we cannot do this.
+According to the [GitHub Marketplace Documentation](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/publishing-actions-in-github-marketplace) - "Each repository must contain a single action."
+Therefore, we will maintain a separate repository for each action that we wish to publish to the marketplace.
+
+The primary goal of combining repositories was to reduce code duplication - this will be accomplished by creating a Node package containing shared code, in an additional repository, to be consumed by the released Actions. This package can easily released using GitHub package registries.
+
+### Build artifact generation
+
+The GitHub actions are built with Typescript - which requires compilation into Javascript.
+This compilation is expected to be done manually by contributors, and checked in alongside pull requests.
+This makes parallel development difficult - when the development branch is updated, one can't just click "Update Branch" as this does not rebuild the Javascript output - it must be done manually.
+Additionally, as dependencies are updated via Dependabot - the build output is not updated, so when a developer does make a change, many unrelated-looking changes can end up in their `dist/index.js`
+
+All of this just goes to show that we need an automated solution:
+
+* Development will occur on the default branch
+* The build output `dist/index.js` will _not_ be checked into this development branch
+* CI for PRs and on schedule will build the sources before running the tests
+* A separate branch, `release`, will be created. Every time a new commit is pushed to `master`, a GitHub Action will be triggered that builds `master` and pushes the resulting build artifact to `release`
```

Comment:
Is considering moving away from `master` branch name to perhaps `main` a possibility at this time?

Reply:
I can build that into this design doc